vault-agent/variables.tf

```diff
@@ -0,0 +1,50 @@
+variable "install_dependencies" {
+  description = "Whether to install/download application dependencies."
+  type        = bool
+}
+
+variable "vault_agent" {
+  description = "Configuration for Vault Agent"
+  type = object({
+    auth_method = object({
+      config = object({
+        role_id   = string
+        secret_id = string
+      })
+    })
+    vault_address   = string
+    vault_ca_cert   = string
+    extra_config    = string
+    release_version = string
+  })
+  default = {
+    auth_method = {
+      config = {
+        role_id   = ""
+        secret_id = ""
+      }
+    }
+    vault_address   = ""
+    vault_ca_cert   = ""
+    extra_config    = ""
+    release_version = "1.17.2"
+  }
+}
+
+variable "external_templates" {
+  description = "List of templates provided by external services (e.g., Fluent Bit)"
+  type = list(object({
+    source_path      = string
+    destination_path = string
+    secret_path      = string
+    secret_key       = string
+    command          = optional(string, "")
+  }))
+  default = []
+}
```

Comment:
I would expect those templates are not to be centralised here, but rather defined in each cloudinit submodules that handles the secret.

For example, the templates for the secrets that "configurations-auto-updater" will consume should be declared in the "configurations-auto-updater" cloudinit submodule.